ads/inabox/inabox-messaging-host.js

```diff
@@ -224,12 +224,20 @@ export class InaboxMessagingHost {
       return this.iframeMap_[sentinel];
     }
 
-    // Walk up on the window tree until find host's direct child window
-    while (source.parent !== this.win_ && source !== this.win_.top) {
-      source = source.parent;
+    // Walk up on the window tree and verify that there is no cross-domain
+    // frame in between the source of the postMessage and the host doc.
+    // If there is, the host doc will not be able to accurately measure
+    // the creative's positions in the page, so return null.
+    let tempWin = source.parent;
+    while (tempWin !== this.win_ && tempWin !== this.win_.top) {
+      if (tempWin.location.origin !== this.win_.location.origin) {
+        return null;
```

Comment:
Can we dev().warn here?  Typically we've determined if a frame is cross by attempting to access state within it as opposed to looking at origin.